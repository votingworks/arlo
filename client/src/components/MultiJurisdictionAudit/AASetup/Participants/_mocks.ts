import { readFileSync } from 'fs'
import { join } from 'path'

const jurisdictionFile = new File(
  [
    readFileSync(
      join(
        __dirname,
        '../../../../../public/sample_jurisdiction_filesheet.csv'
      ),
      'utf8'
    ),
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
