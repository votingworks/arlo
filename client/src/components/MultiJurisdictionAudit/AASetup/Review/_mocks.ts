import { IAuditSettings } from '../../../../types'

export const settingsMock: {
  [key in 'empty' | 'full' | 'offline' | 'batch']: IAuditSettings
} = {
  empty: {
    state: null,
    electionName: null,
    online: null,
    randomSeed: null,
    riskLimit: null,
    auditType: 'BALLOT_POLLING',
    auditName: 'Test Audit',
  },
  full: {
    state: 'AL',
    electionName: 'Election Name',
    online: true,
    randomSeed: '12345',
    riskLimit: 10,
    auditType: 'BALLOT_POLLING',
    auditName: 'Test Audit',
  },
  offline: {
    state: 'AL',
    electionName: 'Election Name',
    online: false,
    randomSeed: '12345',
    riskLimit: 10,
    auditType: 'BALLOT_POLLING',
    auditName: 'Test Audit',
  },
  batch: {
    state: 'AL',
    electionName: 'Election Name',
    online: false,
    randomSeed: '12345',
    riskLimit: 10,
    auditType: 'BATCH_COMPARISON',
    auditName: 'Test Audit',
  },
}

export const sampleSizeMock = {
  sampleSizes: {
    'contest-id': [
      { prob: 0.54, size: 46, key: 'asn' },
      { prob: 0.7, size: 67, key: '0.7' },
      { prob: 0.5, size: 88, key: '0.5' },
      { prob: 0.9, size: 125, key: '0.9' },
    ],
  },
}

export default settingsMock
