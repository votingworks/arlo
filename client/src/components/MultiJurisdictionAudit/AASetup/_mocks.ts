import { ElementType } from '../../../types'
import { setupStages, stageTitles } from './index'
import { ISidebarMenuItem } from '../../Atoms/Sidebar'

const relativeStages = (
  stage: ElementType<typeof setupStages>,
  state: ISidebarMenuItem['state'] = 'live'
): {
  prevStage: ISidebarMenuItem
  nextStage: ISidebarMenuItem
  menuItems: ISidebarMenuItem[]
} => {
  const prevId = setupStages[setupStages.indexOf(stage) - 1]
  const prevStage: ISidebarMenuItem = {
    id: prevId,
    title: stageTitles[prevId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const nextId = setupStages[setupStages.indexOf(stage) + 1]
  const nextStage: ISidebarMenuItem = {
    id: nextId,
    title: stageTitles[nextId],
    active: false,
    activate: jest.fn(),
    state,
  }
  const menuItems = setupStages.map((s: ElementType<typeof setupStages>) => ({
    id: s,
    title: stageTitles[s],
    active: s === stage,
    activate: jest.fn(),
    state,
  }))
  return { prevStage, nextStage, menuItems }
}

export default relativeStages
