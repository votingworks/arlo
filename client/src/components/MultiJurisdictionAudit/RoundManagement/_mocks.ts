export interface INullResultValues {
  [contestId: string]: {
    [choiceId: string]: null | string | number
  }
}

export const resultsMocks: {
  [key in 'emptyInitial' | 'complete']: INullResultValues
} = {
  emptyInitial: {
    'contest-id-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'contest-id-2': {
      'choice-id-3': null,
      'choice-id-4': null,
    },
  },
  complete: {
    'contest-id-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'contest-id-2': {
      'choice-id-3': 1,
      'choice-id-4': 2,
    },
  },
}

export default resultsMocks
