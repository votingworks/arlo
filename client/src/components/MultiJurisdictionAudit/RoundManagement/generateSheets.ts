import jsPDF from 'jspdf'
import { IAuditBoard } from '../useAuditBoards'
import { IBallot } from './useBallots'

export const downloadLabels = async (
  roundNum: number,
  ballots: IBallot[],
  jurisdictionName: string,
  auditName: string
): Promise<string> => {
  /* istanbul ignore else */
  if (ballots.length) {
    const getX = (l: number): number => (l % 3) * 60 + 9 * ((l % 3) + 1)
    const getY = (l: number): number[] => [
      Math.floor(l / 3) * 25.5 + 20,
      Math.floor(l / 3) * 25.5 + 25,
      Math.floor(l / 3) * 25.5 + 34,
    ]
    const labels = new jsPDF({ format: 'letter' })
    labels.setFontSize(9)
    let labelCount = 0
    ballots.forEach(ballot => {
      labelCount += 1
      if (labelCount > 30) {
        labels.addPage('letter')
        labelCount = 1
      }
      const x = getX(labelCount - 1)
      const y = getY(labelCount - 1)
      labels.text(
        labels.splitTextToSize(ballot.auditBoard!.name, 60)[0],
        x,
        y[0]
      )
      labels.text(
        labels.splitTextToSize(`Batch Name: ${ballot.batch.name}`, 60),
        x,
        y[1]
      )
      labels.text(`Ballot Number: ${ballot.position}`, x, y[2])
    })
    labels.autoPrint()
    labels.save(
      `Round ${roundNum} Labels - ${jurisdictionName} - ${auditName}.pdf`
    )
    return labels.output() // returned for test snapshots
  }
  return ''
}

export const downloadPlaceholders = async (
  roundNum: number,
  ballots: IBallot[],
  jurisdictionName: string,
  auditName: string
): Promise<string> => {
  /* istanbul ignore else */
  if (ballots.length) {
    const placeholders = new jsPDF({ format: 'letter' })
    placeholders.setFontSize(20)
    let pageCount = 0
    ballots.forEach(ballot => {
      if (pageCount > 0) placeholders.addPage('letter')
      placeholders.text(
        placeholders.splitTextToSize(ballot.auditBoard!.name, 180),
        20,
        20
      )
      placeholders.text(
        placeholders.splitTextToSize(`Batch Name: ${ballot.batch.name}`, 180),
        20,
        40
      )
      placeholders.text(`Ballot Number: ${ballot.position}`, 20, 100)
      pageCount += 1
    })
    placeholders.autoPrint()
    placeholders.save(
      `Round ${roundNum} Placeholders - ${jurisdictionName} - ${auditName}.pdf`
    )
    return placeholders.output() // returned for test snapshots
  }
  return ''
}

export const downloadAuditBoardCredentials = (
  auditBoards: IAuditBoard[],
  jurisdictionName: string,
  auditName: string
): string => {
  const auditBoardsWithoutBallots: string[] = []
  const auditBoardCreds = new jsPDF({ format: 'letter' })
  auditBoards.forEach((board, i) => {
    const qr: HTMLCanvasElement | null = document.querySelector(
      `#qr-${board.passphrase} > canvas`
    )
    /* istanbul ignore next */
    if (!qr) return
    if (board.currentRoundStatus.numSampledBallots > 0) {
      if (i > 0) auditBoardCreds.addPage('letter')
      const url = qr.toDataURL()
      auditBoardCreds.setFontSize(22)
      auditBoardCreds.setFontStyle('bold')
      auditBoardCreds.text(board.name, 20, 20)
      auditBoardCreds.setFontSize(14)
      auditBoardCreds.setFontStyle('normal')
      auditBoardCreds.text(
        'Scan this QR code to enter the votes you see on your assigned ballots.',
        20,
        40
      )
      auditBoardCreds.addImage(url, 'JPEG', 20, 50, 50, 50)
      auditBoardCreds.text(
        auditBoardCreds.splitTextToSize(
          'If you are not able to scan the QR code, you may also type the following URL into a web browser to access the data entry portal.',
          180
        ),
        20,
        120
      )
      auditBoardCreds.text(
        auditBoardCreds.splitTextToSize(
          `${window.location.origin}/auditboard/${board.passphrase}`,
          180
        ),
        20,
        140
      )
      auditBoardCreds.setTextColor(0, 0, 238)
      auditBoardCreds.textWithLink(
        'Or you can click here to go there directly.',
        20,
        160,
        { url: `${window.location.origin}/auditboard/${board.passphrase}` }
      )
      auditBoardCreds.setTextColor(0, 0, 0)
    } else {
      auditBoardsWithoutBallots.push(board.name)
    }
  })
  if (auditBoardsWithoutBallots.length) {
    auditBoardCreds.addPage('letter')
    auditBoardsWithoutBallots.forEach((name, i) => {
      auditBoardCreds.text(`${name}: No ballots`, 20, i * 10 + 20)
    })
  }
  auditBoardCreds.autoPrint()
  auditBoardCreds.save(
    `Audit Board Credentials - ${jurisdictionName} - ${auditName}.pdf`
  )
  return auditBoardCreds.output() // returned for test snapshots
}
