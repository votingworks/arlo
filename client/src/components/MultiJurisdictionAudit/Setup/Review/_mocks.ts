export const settingsMock = {
  empty: [
    {
      state: null,
      electionName: null,
      online: null,
      randomSeed: null,
      riskLimit: null,
      auditType: 'BALLOT_POLLING',
    },
    // here for type completion but not used in this context
    /* istanbul ignore next */
    async () => true,
  ],
  full: [
    {
      state: 'AL',
      electionName: 'Election Name',
      online: true,
      randomSeed: '12345',
      riskLimit: 10,
      auditType: 'BALLOT_POLLING',
    },
    // here for type completion but not used in this context
    /* istanbul ignore next */
    async () => true,
  ],
  offline: [
    {
      state: 'AL',
      electionName: 'Election Name',
      online: false,
      randomSeed: '12345',
      riskLimit: 10,
      auditType: 'BALLOT_POLLING',
    },
    // here for type completion but not used in this context
    /* istanbul ignore next */
    async () => true,
  ],
}

export const sampleSizeMock = {
  sampleSizes: {
    'contest-id': [
      { prob: 0.54, size: 46, key: 'asn' },
      { prob: 0.7, size: 67, key: '0.7' },
      { prob: 0.5, size: 88, key: '0.5' },
      { prob: 0.9, size: 125, key: '0.9' },
    ],
  },
}

export default settingsMock
