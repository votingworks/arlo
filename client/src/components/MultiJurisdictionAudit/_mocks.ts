import { contestMocks } from './AASetup/Contests/_mocks'
import { IFileInfo } from './useJurisdictions'
import {
  auditSettings,
  manifestFile,
  talliesFile,
} from './useSetupMenuItems/_mocks'
import { FileProcessingStatus } from './useSetupMenuItems/getJurisdictionFileStatus'
import { IAuditSettings } from '../../types'
import { IBallot } from './RoundManagement/useBallots'
import { IAuditBoard } from './useAuditBoards'
import { IRound } from './useRoundsAuditAdmin'

const manifestFormData: FormData = new FormData()
manifestFormData.append('manifest', manifestFile, manifestFile.name)
const talliesFormData: FormData = new FormData()
talliesFormData.append('batchTallies', talliesFile, talliesFile.name)

export const jaApiCalls = {
  getUser: {
    url: '/api/me',
    response: {
      type: 'jurisdiction_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          election: {
            id: '1',
            auditName: 'audit one',
            electionName: 'election one',
            state: 'AL',
          },
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          election: {
            id: '2',
            auditName: 'audit two',
            electionName: 'election two',
            state: 'AL',
          },
        },
        {
          id: 'jurisdiction-id-3',
          name: 'Jurisdiction Three',
          election: {
            id: '1',
            auditName: 'audit one',
            electionName: 'election one',
            state: 'AL',
          },
        },
      ],
      organizations: [],
    },
  },
  getRounds: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round',
    response: { rounds: [] },
  },
  getBallotManifestFile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/ballot-manifest',
    response,
  }),
  getBatchTalliesFile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/batch-tallies',
    response,
  }),
  getCVRSfile: (response: IFileInfo) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/cvrs',
    response,
  }),
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/settings',
    response,
  }),
  putManifest: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/ballot-manifest',
    options: {
      method: 'PUT',
      body: manifestFormData,
    },
    response: { status: 'ok' },
  },
  putTallies: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/batch-tallies',
    options: {
      method: 'PUT',
      body: talliesFormData,
    },
    response: { status: 'ok' },
  },
  getAuditBoards: (auditBoards: IAuditBoard[]) => ({
    url:
      '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/audit-board',
    response: { auditBoards },
  }),
  getBallots: (ballots: IBallot[]) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/ballots',
    response: { ballots },
  }),
}

export const aaApiCalls = {
  getUser: {
    url: '/api/me',
    response: {
      type: 'audit_admin',
      name: 'Joe',
      email: 'test@email.org',
      jurisdictions: [],
      organizations: [
        {
          id: 'org-id',
          name: 'State of California',
          elections: [],
        },
      ],
    },
  },
  getRounds: (rounds: IRound[]) => ({
    url: '/api/election/1/round',
    response: { rounds },
  }),
  getJurisdictions: {
    url: '/api/election/1/jurisdiction',
    response: {
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          ballotManifest: { file: null, processing: null },
          currentRoundStatus: null,
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          ballotManifest: { file: null, processing: null },
          currentRoundStatus: null,
        },
      ],
    },
  },
  getBatchJurisdictions: {
    url: '/api/election/1/jurisdiction',
    response: {
      jurisdictions: [
        {
          id: 'jurisdiction-id-1',
          name: 'Jurisdiction One',
          ballotManifest: { file: null, processing: null },
          batchTallies: { file: null, processing: null },
          currentRoundStatus: null,
        },
        {
          id: 'jurisdiction-id-2',
          name: 'Jurisdiction Two',
          ballotManifest: { file: null, processing: null },
          batchTallies: { file: null, processing: null },
          currentRoundStatus: null,
        },
      ],
    },
  },
  getJurisdictionFile: {
    url: '/api/election/1/jurisdiction/file',
    response: {
      file: {
        contents: null,
        name: 'file name',
        uploadedAt: 'a long time ago in a galaxy far far away',
      },
      processing: {
        status: FileProcessingStatus.Processed,
        error: null,
        startedAt: 'once upon a time',
        endedAt: 'and they lived happily ever after',
      },
    },
  },
  getContests: {
    url: '/api/election/1/contest',
    response: contestMocks.filledTargeted,
  },
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/settings',
    response,
  }),
  putSettings: {
    url: '/api/election/1/settings',
    options: {
      method: 'PUT',
      body: JSON.stringify(auditSettings.all),
      headers: { 'Content-Type': 'application/json' },
    },
    response: { status: 'ok' },
  },
  getSampleSizes: {
    url: '/api/election/1/sample-sizes',
    response: { sampleSizes: null },
  },
}
