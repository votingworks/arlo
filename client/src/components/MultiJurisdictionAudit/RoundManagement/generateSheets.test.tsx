import React from 'react'
import { render } from '@testing-library/react'
import { auditBoardMocks } from '../_mocks'
import QRs from './QRs'
import { downloadDataEntry } from './generateSheets'
import { IAuditBoard } from '../useAuditBoards'

jest.mock('jspdf', () => {
  const realjspdf = jest.requireActual('jspdf')
  const mockjspdf = new realjspdf({ format: 'letter' })
  // eslint-disable-next-line func-names
  return function() {
    return {
      ...mockjspdf,
      addImage: jest.fn(),
    }
  }
})

describe('generateSheets', () => {
  it('generates data entry sheets', () => {
    window.URL.createObjectURL = jest.fn()
    render(
      <QRs
        passphrases={auditBoardMocks.double.map(
          (b: IAuditBoard) => b.passphrase
        )}
      />
    )
    const pdf = downloadDataEntry(auditBoardMocks.double)
      .replace(/CreationDate \([^)]+\)/g, '') // remove the timestamp
      .replace(/ID \[[^\]]+\]/g, '') // remove the unique id
    expect(pdf).toMatchSnapshot() // test the rest of the file now it's deterministic
  })

  it('generates data entry sheets with ballotless audit board', () => {
    window.URL.createObjectURL = jest.fn()
    render(
      <QRs
        passphrases={auditBoardMocks.noBallots.map(
          (b: IAuditBoard) => b.passphrase
        )}
      />
    )
    const pdf = downloadDataEntry(auditBoardMocks.noBallots)
      .replace(/CreationDate \([^)]+\)/g, '') // remove the timestamp
      .replace(/ID \[[^\]]+\]/g, '') // remove the unique id
    expect(pdf).toMatchSnapshot() // test the rest of the file now it's deterministic
  })
})
