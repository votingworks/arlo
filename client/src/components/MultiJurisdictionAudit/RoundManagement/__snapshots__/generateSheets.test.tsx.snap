// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSheets generates data entry sheets 1`] = `
"%PDF-1.3
%ºß¬à
3 0 obj
<</Type /Page
/Parent 1 0 R
/Resources 2 0 R
/MediaBox [0 0 612.00 792.00]
/Contents 4 0 R
>>
endobj
4 0 obj
<<
/Length 479
>>
stream
0.57 w
0 G
BT
/F2 22 Tf
25.30 TL
0 g
56.69 735.31 Td
(Audit Board #01) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 678.61 Td
(Scan this QR code to enter the votes you see on your assigned ballots.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 451.84 Td
(If you are not able to scan the QR code, you may also type the following URL into) Tj
T* (a web browser to access the data entry portal.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 395.15 Td
(http://localhost/auditboard/happy-randomness) Tj
ET
endstream
endobj
5 0 obj
<</Type /Page
/Parent 1 0 R
/Resources 2 0 R
/MediaBox [0 0 612.00 792.00]
/Contents 6 0 R
>>
endobj
6 0 obj
<<
/Length 489
>>
stream
0.57 w
0 G
BT
/F2 22 Tf
25.30 TL
0 g
56.69 735.31 Td
(Audit Board #02) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 678.61 Td
(Scan this QR code to enter the votes you see on your assigned ballots.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 451.84 Td
(If you are not able to scan the QR code, you may also type the following URL into) Tj
T* (a web browser to access the data entry portal.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 395.15 Td
(http://localhost/auditboard/happy-secondary-randomness) Tj
ET
endstream
endobj
1 0 obj
<</Type /Pages
/Kids [3 0 R 5 0 R ]
/Count 2
>>
endobj
7 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
8 0 obj
<<
/Type /Font
/BaseFont /Helvetica-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
9 0 obj
<<
/Type /Font
/BaseFont /Helvetica-Oblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
10 0 obj
<<
/Type /Font
/BaseFont /Helvetica-BoldOblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
11 0 obj
<<
/Type /Font
/BaseFont /Courier
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
12 0 obj
<<
/Type /Font
/BaseFont /Courier-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
13 0 obj
<<
/Type /Font
/BaseFont /Courier-Oblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
14 0 obj
<<
/Type /Font
/BaseFont /Courier-BoldOblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
15 0 obj
<<
/Type /Font
/BaseFont /Times-Roman
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
16 0 obj
<<
/Type /Font
/BaseFont /Times-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
17 0 obj
<<
/Type /Font
/BaseFont /Times-Italic
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
18 0 obj
<<
/Type /Font
/BaseFont /Times-BoldItalic
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
19 0 obj
<<
/Type /Font
/BaseFont /ZapfDingbats
/Subtype /Type1
/FirstChar 32
/LastChar 255
>>
endobj
20 0 obj
<<
/Type /Font
/BaseFont /Symbol
/Subtype /Type1
/FirstChar 32
/LastChar 255
>>
endobj
2 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/Font <<
/F1 7 0 R
/F2 8 0 R
/F3 9 0 R
/F4 10 0 R
/F5 11 0 R
/F6 12 0 R
/F7 13 0 R
/F8 14 0 R
/F9 15 0 R
/F10 16 0 R
/F11 17 0 R
/F12 18 0 R
/F13 19 0 R
/F14 20 0 R
>>
/XObject <<
>>
>>
endobj
21 0 obj
<<
/S /Named
/Type /Action
/N /Print
>>
endobj
22 0 obj
<<
/Producer (jsPDF 1.5.3)
/
>>
endobj
23 0 obj
<<
/Type /Catalog
/Pages 1 0 R
/OpenAction [3 0 R /FitH null]
/PageLayout /OneColumn
/OpenAction 21 0 R
>>
endobj
xref
0 24
0000000000 65535 f 
0000001303 00000 n 
0000003128 00000 n 
0000000015 00000 n 
0000000124 00000 n 
0000000654 00000 n 
0000000763 00000 n 
0000001366 00000 n 
0000001491 00000 n 
0000001621 00000 n 
0000001754 00000 n 
0000001892 00000 n 
0000002016 00000 n 
0000002145 00000 n 
0000002277 00000 n 
0000002413 00000 n 
0000002541 00000 n 
0000002668 00000 n 
0000002797 00000 n 
0000002930 00000 n 
0000003032 00000 n 
0000003378 00000 n 
0000003434 00000 n 
0000003520 00000 n 
trailer
<<
/Size 24
/Root 23 0 R
/Info 22 0 R
/
>>
startxref
3643
%%EOF"
`;

exports[`generateSheets generates data entry sheets with ballotless audit board 1`] = `
"%PDF-1.3
%ºß¬à
3 0 obj
<</Type /Page
/Parent 1 0 R
/Resources 2 0 R
/MediaBox [0 0 612.00 792.00]
/Contents 4 0 R
>>
endobj
4 0 obj
<<
/Length 479
>>
stream
0.57 w
0 G
BT
/F2 22 Tf
25.30 TL
0 g
56.69 735.31 Td
(Audit Board #01) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 678.61 Td
(Scan this QR code to enter the votes you see on your assigned ballots.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 451.84 Td
(If you are not able to scan the QR code, you may also type the following URL into) Tj
T* (a web browser to access the data entry portal.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 395.15 Td
(http://localhost/auditboard/happy-randomness) Tj
ET
endstream
endobj
5 0 obj
<</Type /Page
/Parent 1 0 R
/Resources 2 0 R
/MediaBox [0 0 612.00 792.00]
/Contents 6 0 R
>>
endobj
6 0 obj
<<
/Length 489
>>
stream
0.57 w
0 G
BT
/F2 22 Tf
25.30 TL
0 g
56.69 735.31 Td
(Audit Board #02) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 678.61 Td
(Scan this QR code to enter the votes you see on your assigned ballots.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 451.84 Td
(If you are not able to scan the QR code, you may also type the following URL into) Tj
T* (a web browser to access the data entry portal.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 395.15 Td
(http://localhost/auditboard/happy-secondary-randomness) Tj
ET
endstream
endobj
7 0 obj
<</Type /Page
/Parent 1 0 R
/Resources 2 0 R
/MediaBox [0 0 612.00 792.00]
/Contents 8 0 R
>>
endobj
8 0 obj
<<
/Length 489
>>
stream
0.57 w
0 G
BT
/F2 22 Tf
25.30 TL
0 g
56.69 735.31 Td
(Audit Board #02) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 678.61 Td
(Scan this QR code to enter the votes you see on your assigned ballots.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 451.84 Td
(If you are not able to scan the QR code, you may also type the following URL into) Tj
T* (a web browser to access the data entry portal.) Tj
ET
BT
/F1 14 Tf
16.10 TL
0 g
56.69 395.15 Td
(http://localhost/auditboard/happy-secondary-randomness) Tj
ET
endstream
endobj
9 0 obj
<</Type /Page
/Parent 1 0 R
/Resources 2 0 R
/MediaBox [0 0 612.00 792.00]
/Contents 10 0 R
>>
endobj
10 0 obj
<<
/Length 88
>>
stream
0.57 w
0 G
BT
/F1 14 Tf
16.10 TL
0 g
56.69 735.31 Td
(Audit Board #01: No ballots) Tj
ET
endstream
endobj
1 0 obj
<</Type /Pages
/Kids [3 0 R 5 0 R 7 0 R 9 0 R ]
/Count 4
>>
endobj
11 0 obj
<<
/Type /Font
/BaseFont /Helvetica
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
12 0 obj
<<
/Type /Font
/BaseFont /Helvetica-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
13 0 obj
<<
/Type /Font
/BaseFont /Helvetica-Oblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
14 0 obj
<<
/Type /Font
/BaseFont /Helvetica-BoldOblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
15 0 obj
<<
/Type /Font
/BaseFont /Courier
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
16 0 obj
<<
/Type /Font
/BaseFont /Courier-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
17 0 obj
<<
/Type /Font
/BaseFont /Courier-Oblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
18 0 obj
<<
/Type /Font
/BaseFont /Courier-BoldOblique
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
19 0 obj
<<
/Type /Font
/BaseFont /Times-Roman
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
20 0 obj
<<
/Type /Font
/BaseFont /Times-Bold
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
21 0 obj
<<
/Type /Font
/BaseFont /Times-Italic
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
22 0 obj
<<
/Type /Font
/BaseFont /Times-BoldItalic
/Subtype /Type1
/Encoding /WinAnsiEncoding
/FirstChar 32
/LastChar 255
>>
endobj
23 0 obj
<<
/Type /Font
/BaseFont /ZapfDingbats
/Subtype /Type1
/FirstChar 32
/LastChar 255
>>
endobj
24 0 obj
<<
/Type /Font
/BaseFont /Symbol
/Subtype /Type1
/FirstChar 32
/LastChar 255
>>
endobj
2 0 obj
<<
/ProcSet [/PDF /Text /ImageB /ImageC /ImageI]
/Font <<
/F1 11 0 R
/F2 12 0 R
/F3 13 0 R
/F4 14 0 R
/F5 15 0 R
/F6 16 0 R
/F7 17 0 R
/F8 18 0 R
/F9 19 0 R
/F10 20 0 R
/F11 21 0 R
/F12 22 0 R
/F13 23 0 R
/F14 24 0 R
>>
/XObject <<
>>
>>
endobj
25 0 obj
<<
/S /Named
/Type /Action
/N /Print
>>
endobj
26 0 obj
<<
/S /Named
/Type /Action
/N /Print
>>
endobj
27 0 obj
<<
/Producer (jsPDF 1.5.3)
/
>>
endobj
28 0 obj
<<
/Type /Catalog
/Pages 1 0 R
/OpenAction [3 0 R /FitH null]
/PageLayout /OneColumn
/OpenAction 25 0 R
/OpenAction 26 0 R
>>
endobj
xref
0 29
0000000000 65535 f 
0000002201 00000 n 
0000004041 00000 n 
0000000015 00000 n 
0000000124 00000 n 
0000000654 00000 n 
0000000763 00000 n 
0000001303 00000 n 
0000001412 00000 n 
0000001952 00000 n 
0000002062 00000 n 
0000002276 00000 n 
0000002402 00000 n 
0000002533 00000 n 
0000002667 00000 n 
0000002805 00000 n 
0000002929 00000 n 
0000003058 00000 n 
0000003190 00000 n 
0000003326 00000 n 
0000003454 00000 n 
0000003581 00000 n 
0000003710 00000 n 
0000003843 00000 n 
0000003945 00000 n 
0000004294 00000 n 
0000004350 00000 n 
0000004406 00000 n 
0000004492 00000 n 
trailer
<<
/Size 29
/Root 28 0 R
/Info 27 0 R
/
>>
startxref
4634
%%EOF"
`;
